reconcile/utils/saasherder.py

```diff
@@ -859,6 +859,91 @@ def update_moving_commit(self, job_spec):
             f"{namespace_name}/{env_name}/{ref}"
         self.state.add(key, value=commit_sha, force=True)
 
+    def get_upstream_jobs_diff(self, dry_run, current_state):
+        results = threaded.run(self.get_upstream_jobs_diff_saas_file,
+                               self.saas_files,
+                               self.thread_pool_size,
+                               dry_run=dry_run,
+                               current_state=current_state)
+        return [item for sublist in results for item in sublist]
+
+    def get_upstream_jobs_diff_saas_file(self, saas_file, dry_run,
+                                         current_state):
+        saas_file_name = saas_file['name']
+        timeout = saas_file.get('timeout') or None
+        pipelines_provider = self._get_pipelines_provider(saas_file)
+        trigger_specs = []
+        for rt in saas_file['resourceTemplates']:
+            rt_name = rt['name']
+            for target in rt['targets']:
```

Comment:
Can we move the body of this loop into its own method for the sake of the tests to come?

Reply:
this is common around this class and should be handled once for all repetitions. feel free to take a look around and suggest better approaches through PRs.